cobalt/renderer/cobalt_render_frame_observer_browsertest.cc

```diff
@@ -0,0 +1,41 @@
+// Copyright 2025 The Cobalt Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "cobalt/renderer/cobalt_render_frame_observer.h"
+
+#include "content/public/common/isolated_world_ids.h"
+#include "content/public/test/render_view_test.h"
+#include "third_party/blink/public/web/blink.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+#include "v8/include/v8.h"
+
+namespace cobalt {
+
+class CobaltRenderFrameObserverTest : public content::RenderViewTest {
+ public:
+  content::RenderFrame* render_frame() { return GetMainRenderFrame(); }
+  blink::WebLocalFrame* web_local_frame() { return GetMainFrame(); }
+};
+
+TEST_F(CobaltRenderFrameObserverTest, InstallCloseAndMinimizeTest) {
+  int close_or_minimize_call_count = 0;
+  new CobaltRenderFrameObserver(render_frame(),
+                                base::BindRepeating(
+                                    [](int* close_or_minimize_call_count) {
+                                      (*close_or_minimize_call_count)++;
+                                    },
+                                    &close_or_minimize_call_count));
+  EXPECT_EQ(0, close_or_minimize_call_count);
+  ExecuteJavaScriptForTests("window.close();");
+  EXPECT_EQ(1, close_or_minimize_call_count);
+  ExecuteJavaScriptForTests("window.minimize();");
+  EXPECT_EQ(2, close_or_minimize_call_count);
+  int user_on_exit_strategy = -1;
+  EXPECT_TRUE(ExecuteJavaScriptAndReturnIntValue(
+      u"window.h5vcc.system.userOnExitStrategy", &user_on_exit_strategy));
+  EXPECT_GE(user_on_exit_strategy, 0);
+  EXPECT_LE(user_on_exit_strategy, 2);
```

Comment:
If you take up my suggestion of having an enum class for the exit strategies in Configuration, then you can add a kCount entry at the end and use it here?